src/roman-numerals/index.js

```diff
@@ -4,6 +4,35 @@
  * @param {string} roman The all-caps Roman numeral between 1 and 3999 (inclusive).
  * @returns {number} The decimal equivalent.
  */
-function romanToDecimal(roman) {}
+function romanToDecimal(roman) {
+    let reg = /MMMM|CCCC|IIII|DDDD|XXXX|LLLL|VVVV/;
+    const charInt = char=>{
+        switch (char) {
+            case 'I': return 1;
+            case 'V': return 5;
+            case 'X': return 10;
+            case 'L': return 50;
+            case 'C': return 100;
+            case 'D': return 500;
+            case 'M': return 1000;
+            default: return -1;
+        }
+    }
+    let val = charInt(roman.charAt(0));
+    let num, prevNum
+    if (reg.test(roman) || roman===null) {
+        val = -1;
```

Comment:
Why?

The test for null should be separate and different.

Also, why are you you specifically testing `VVVV, DDDD,...` etc?

Reply:
I will work on modifying that